share/completions/wpctl.fish

```diff
@@ -0,0 +1,51 @@
+set -l commands status get-volume inspect set-default set-volume set-mute set-profile clear-default
+
+function __wpctl_get_nodes -a section -a type
+    for node in (string match -r " ├─ $type:.*(?:\n │.*)+" "$(string match -r "$section*(?:\n [├│└].*)*" "$(wpctl status)")" | string match -rg '^ │[ \*]{6}([^\[]*)')
```

Comment:
This is pretty clever, but in a way that tells me there should be a better way to get this information - `wpctl status` isn't made for parsing.

Case in point: This doesn't work for me. I run `__wpctl_get_nodes Audio Sources` and it doesn't return anything.

This is because `string match -r "$section*(?:\n [├│└].*)*" "$(wpctl status)"` only matches the string "Audio".

Instead, I would try it with a while-read-loop like this:

```fish
set -l havesection
set -l havetype
wpctl status | while read -l line
    if set -q havesection[1]
        test -z "$line"; and break
        if set -q havetype[1]
            string match -rq '^\s*\│\s*$' -- $line; and break
            echo $line
        else
            string match -q "*$type*" -- $line
            and set havetype 1
        end
    else
        string match -q "$section*" -- $line
        and set havesection 1
    end
end
```

Reply:
Thanks for the write-up